rlbottraining/common_graders/rl_graders.py

```diff
@@ -15,27 +16,149 @@
 from rlbottraining.common_graders.compound_grader import CompoundGrader
 from rlbottraining.common_graders.timeout import FailOnTimeout, PassOnTimeout
 from rlbottraining.common_graders.goal_grader import PassOnGoalForAllyTeam
+from rlbot.training.training import Pass, Fail, Grade
+import copy
 
 
 class RocketLeagueStrikerGrader(CompoundGrader):
     """
     A Grader which aims to match the striker training.
     """
 
-    def __init__(self, timeout_seconds=4.0, ally_team=0):
+    def __init__(self, timeout_seconds=4.0, ally_team=0, timeout_override=False, ground_override=False):
+        self.timeout_override = timeout_override
+        self.ground_override = ground_override
         super().__init__([
             PassOnGoalForAllyTeam(ally_team),
-            FailOnBallOnGroundAfterTimeout(timeout_seconds),
+            FailOnBallOnGround(),
+            FailOnTimeout(timeout_seconds),
         ])
 
-class FailOnBallOnGroundAfterTimeout(FailOnTimeout):
+    def on_tick(self, tick: TrainingTickPacket) -> Optional[Grade]:
+        grades = [grader.on_tick(tick) for grader in self.graders]
+        return self.grade_chooser(grades)
+
+    def grade_chooser(self, grades) -> Optional[Grade]:
+        """
+        Chooses the importance of the grades
+        """
```

Comment:
```suggestion
        # Choose the importance of the grades.
```
I think this is better as there's no re-use and a lot of assumptions about the order of `grades`.